src/bench/bip324_ecdh.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+
+#include <key.h>
+#include <pubkey.h>
+#include <random.h>
+#include <secp256k1.h>
+#include <secp256k1_ellswift.h>
+
+#include <cstddef>
+
+CKey GetRandomKey()
+{
+    CKey key;
+    key.MakeNewKey(true);
+    return key;
+}
+
+void GetEll64(const CPubKey& pubkey, unsigned char* ell64) {
+    auto ctx = GetVerifyContext();
+    secp256k1_pubkey pubkey_internal;
+    if (!secp256k1_ec_pubkey_parse(ctx, &pubkey_internal, pubkey.data(), pubkey.size())) {
+        return;
+    }
+
+    std::array<unsigned char, 32> rnd32;
+    GetRandBytes(rnd32);
+    secp256k1_ellswift_encode(ctx, ell64, &pubkey_internal, rnd32.data());
```

Comment:
The new ellswift API has a function `secp256k1_ellswift_create` which goes directly from private key + randomness to ellswift encoding, bypassing the need for encoding the public key in the traditional form.

Reply:
Much nicer. Thank you for the reminder. Done.